### PR TITLE
[2.9-python3-acceptancetest] Yet even more python3 compat fixes for acceptance tests

### PR DESCRIPTION
**Pull request by achilleasa**
Followup to #12516. Also, fixes an incorrect assertion in `assess_persistent_storage`.

### CODE REVIEW

File: acceptancetests/assess_persistent_storage.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def assert_app_removal_msg(client, charm_name):
     # merge_stderr=True is required
     app_removal_output = client.get_juju_output(
         'remove-application', charm_name, '--show-log',
-        include_e=False, merge_stderr=True)
+        include_e=False, merge_stderr=True).decode('utf-8')
```
Comment (SimonRichardson) [inline acceptancetests/assess_persistent_storage.py:168]: Maybe put `.decode('utf-8')` into the `get_juju_output` method?

Comment (achilleasa) [reply acceptancetests/assess_persistent_storage.py:168]: In some cases we pipe that to `json.loads`. It's easier to patch at the call site where we need to run a regex against that.
I think we should just make things work under python3 and slap a big "caveat emptor: use the bash framework for writing new tests" warning in the readme.

Comment (SimonRichardson) [reply acceptancetests/assess_persistent_storage.py:168]: Fine :+1:

**Review by SimonRichardson — APPROVED** [review]

File: acceptancetests/assess_persistent_storage.py (hunk 1 above)
Comment (achilleasa) [reply acceptancetests/assess_persistent_storage.py:168]: Hmmm... ok, let me change that everywhere :-(

**Comment by achilleasa** [comment]
$$merge$$

**Comment by achilleasa** [comment]
$$merge$$ (go snap installation failed)

**Pull request merged**